crates/testing/src/script.rs

```diff
@@ -106,6 +107,11 @@ pub async fn run_test_script<TYPES, S: TaskState<Event = Arc<HotShotEvent<TYPES>
             }
         }
 
+        if let Some(duration) = check_delay {
+            tracing::info!("Sleeping thread for {:?} before validating.", duration);
+            sleep(duration);
+        }
```

Comment:
Is this the change that fixes the nondeterministic issue? I guess we can make the `duration` configurable for each test, or even move the `check_delay` parameter into `state.outputs` to only delay for some specific outputs. But this is a nice fix!